tests_e2e/orchestrator/lib/agent_test_suite.py

```diff
@@ -233,15 +260,21 @@ def execute(self, node: Node, test_suite: List[Type[AgentTest]]) -> None:
                 for r in results:
                     log.info("\t%s", r)
                 log.info("")
-
             finally:
                 self._collect_node_logs()
 
+        except:   # pylint: disable=bare-except
+            # Log the error here so the it is decorated with the thread name, then re-raise
+            log.exception("Test suite failed")
+            raise
+
         finally:
             self._clean_up()
+            current_thread().name = thread_name
 
         # Fail the entire test suite if any test failed
-        assert_that(failed).described_as("One or more tests failed").is_length(0)
+        if len(failed) > 0:
+            fail(f"{[self.context.suite_name]} One or more tests failed: {failed}")
```

Comment:
Using fail() instead of assert)that() improves the formatting of the error message a little bit.